precompiles/assets/types.go

```diff
@@ -42,16 +44,41 @@ func (p Precompile) DepositWithdrawParamsFromInputs(ctx sdk.Context, args []inte
 	}
 	clientChainAddrLength := info.AddressLength
 
-	// the length of client chain address inputted by caller is 32, so we need to check the length and remove the padding according to the actual length.
-	assetAddr, ok := args[1].([]byte)
-	if !ok || assetAddr == nil {
-		return nil, fmt.Errorf(exocmn.ErrContractInputParaOrType, 1, "[]byte", args[1])
-	}
-	// #nosec G115
-	if uint32(len(assetAddr)) < clientChainAddrLength {
-		return nil, fmt.Errorf(exocmn.ErrInvalidAddrLength, len(assetAddr), clientChainAddrLength)
+	switch method.Name {
+	// deposit and withdraw transactions
+	case MethodDepositLST, MethodWithdrawLST:
+		// the length of client chain address inputted by caller is 32, so we need to check the length and remove the padding according to the actual length.
+		assetAddr, ok := args[1].([]byte)
+		if !ok || assetAddr == nil {
+			return nil, fmt.Errorf(exocmn.ErrContractInputParaOrType, 1, "[]byte", args[1])
+		}
+		// #nosec G115
+		if uint32(len(assetAddr)) < clientChainAddrLength {
+			return nil, fmt.Errorf(exocmn.ErrInvalidAddrLength, len(assetAddr), clientChainAddrLength)
+		}
+		depositWithdrawParams.AssetsAddress = assetAddr[:clientChainAddrLength]
+		if method.Name == MethodDepositLST {
+			depositWithdrawParams.Action = assetstypes.DepositLST
+		} else {
+			depositWithdrawParams.Action = assetstypes.WithdrawLST
+		}
+	case MethodDepositNST, MethodWithdrawNST:
+		validatorPubkey, ok := args[1].([]byte)
```

Comment:
Yeah, there is currently no logic to check ValidatorPubkey, because we need to consider compatibility with different client chains. This check should be implemented in the future.